### PR TITLE
Topic recognition

### PR DESCRIPTION
**Pull request by Peiran66**
My student number is 45223499.
The main.py in this branch is the main function of processing data, running the model and drawing, and model.py is the model function and structure used in the model. At the same time, a readme file describing all the processes and results of this project is in this branch.

### CODE REVIEW

**Comment by SiyuLiu0329** [comment]
- Sufficiently detailed commit messages
- Code valid for Normal Difficulty
- Results are good

Other Feedback
- Good
- (OPTIONAL) Would put main logic in `if __name__ == '__main__'` for the sake of following python convention.
- (OPTIONAL) I would break up long functions into reusable chunks when building the model. It makes hyper parameter adjustment much easier.

**Comment by rexsw** [comment]
Code looks good for normal difficulty

For next time;

Tensorflow has functions for image resizing and more so there’s no need to also use PIL and cv2 in your project. 

Paths such as "C:/Users/s4522349/Downloads/ISIC2018_Task1-2_Training_Input_x2/" won’t work for the user and should either be set dynamical or it needs to be made clear to the user that they need to change these paths

Comments such as 
    # add segmentation layers
    def segmentation_layer(x):
Add nothing to the readability of the code and should be avoided. But overall comments are good. 

Code in files such as main.py should be in a main function.

File: recognition/45223499_improved_unet/main.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
+import tensorflow as tf
```
Comment (shakes76) [inline recognition/45223499_improved_unet/main.py:1]: Code header block should be added for all scripts/source files

**Comment by shakes76** [comment]
## TF Usage
OK
## Good Practice (Design/Commenting)
Good design and comments
## Algorithm 
Driver Script present
Shows Usage & Demo & Visualisation & Data usage, Readme OK
Module present
## Commit Log
Meaningful commit messages 
Progressive commits used 
## Documentation
Description and Comments good
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Request Description. In the future, a more detailed pull request could've been done. When developer receiving the request is not known, the request could be rejected. Remember you could use/reuse the README info.

**Pull request merged**